eva-accession-import-dbsnp2/src/main/java/uk/ac/ebi/eva/accession/dbsnp2/io/DbsnpJsonItemProcessor.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2019 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.ebi.eva.accession.dbsnp2.io;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.ac.ebi.eva.accession.dbsnp2.model.DbsnpJson;
+
+import org.springframework.batch.item.ItemProcessor;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class DbsnpJsonItemProcessor implements ItemProcessor<Map<String, Object>, List<DbsnpJson>> {
+
+    private ObjectMapper mapper = new ObjectMapper();
+
+    public List<DbsnpJson> process(Map<String, Object> jsonItem) throws Exception {
+        JsonNode jsonNode = mapper.valueToTree(jsonItem);
+        return parseDbsnpJsonPlacements(jsonNode);
+    }
+
+    /**
+     * Parses the JSON into a list of dbSNP submitted variants
+     * @see <a https://github.com/Ensembl/ensembl-variation/blob/master/scripts/import/dbSNP_v2/load_dbsnp.pl#L331>
+     *     ensembl-import-dbsnp-v2</a>
+     * @param jsonRootNode root node of a dbSNP JSON line item
+     * @return list of dbSNP submitted variants
+     */
+    private List<DbsnpJson> parseDbsnpJsonPlacements(JsonNode jsonRootNode) throws IOException {
+        List<DbsnpJson> dbsnpJsons = new ArrayList<>();
+        DbsnpJson dbsnpJson = new DbsnpJson();
+        dbsnpJson.setClusteredVariantAccession(jsonRootNode.path("refsnp_id").asLong());
```

Comment:
Is asLong null resistant or will this statement explode if there are no REFSNP_IDs?

Reply:
Not to persist any value at all? 

suggest to replace with ```Long.valueOf(jsonRootNode.path("refsnp_id"))```

Reply:
The benefit of Mongo is that you don't need to persist information that is not available, i.e. null, which allows to save space. In the import of dbSNP 1.0 we already avoid persisting the RS ID when it is not present; the code should do it automatically as soon as the input is converted to the appropriate models.